src/openforms/forms/api/serializers.py

```diff
@@ -368,6 +380,26 @@ def validate_backend_options(self, attrs, backend_field, options_field, registry
         # serializer does some normalization, so make sure to update the data
         attrs[options_field] = serializer.data
 
+    def validate_auto_login_backend(self, attrs):
+        field_name = "auto_login_authentication_backend"
+
+        # If an auto login backend is supplied, it must be present in `authentication_backends`
+        if (
+            attrs.get(field_name)
+            and attrs[field_name] not in attrs["authentication_backends"]
+        ):
+            raise serializers.ValidationError(
+                {
+                    field_name: ErrorDetail(
+                        _(
+                            "The `{auto_login_authentication_backend}` must be one of the selected backends "
+                            "from `authentication_backends`"
+                        ).format(auto_login_authentication_backend=field_name),
+                        code="invalid",
+                    )
+                }
+            )
```

Comment:
This can still lead to inconsistent situations on PATCH calls. If you remove available backends but don't touch the `autologin` backend, this validation won't catch it. Instead, you should use `openforms.api.utils.get_from_serializer_data_or_instance` which will compare against the existing record for partial updates. Please also add a test case for this situation.